A3A/addons/core/functions/Missions/fn_LOG_Crashsite.sqf

```diff
@@ -0,0 +1,721 @@
+private _vehicles = [];
+private _groups = [];
+private _effectsAndProps = [];
+
+//choosing enemy delivery target site
+private _potentialSites = outposts + milbases + airportsX;
+_potentialSites = _potentialSites select {sidesX getVariable [_x,sideUnknown] != teamPlayer};
+private _deliverySite = getMarkerPos _markerX;
+if(count _potentialSites > 0) then {
+    //TODO: find closest enemy outposts
+    _randomEnemySite = selectRandom _potentialSites;
+    _deliverySite = getMarkerPos _randomEnemySite;
+};
+
+Info_3("Difficulty: %1, Side: %2, Destination: %3", str _difficult, str _sideX, _destinationName);
+
+//finding crash position
+private _angle = random 360;
+private _countX = 0;
+private _distance = if (_difficult) then {2000} else {3000};
+private _posCrashOrigin = [];
+private _respawnTeamPlayerMarkerPos = getMarkerPos respawnTeamPlayer;
+while {true} do {
+	_posCrashOrigin = _missionOriginPos getPos [_distance,_angle];
+    _respawnTeamPlayerMarkerPos = getMarkerPos respawnTeamPlayer;
+    _outOfBounds = _posCrashOrigin findIf { (_x < 0) || {_x > worldSize}} != -1;
+	if (!surfaceIsWater _posCrashOrigin 
+        && _posCrashOrigin distance _respawnTeamPlayerMarkerPos < 4000 
+        && _posCrashOrigin distance _respawnTeamPlayerMarkerPos > 2000
+        && !_outOfBounds
+    ) exitWith {};
+	_angle = _angle + 1;
+	_countX = _countX + 1;
+	if (_countX > 360) then {
+		_countX = 0;
+		_distance = _distance - 500;
+	};
+};
+
+// selecting classnames
+private _reconvehicleClass = selectRandom ((_faction get "vehiclesPlanesTransport") + (_faction get "uavsAttack") + (_faction get "vehiclesDropPod"));
+private _pilotClass = _faction get "unitPilot";
+
+if (_reconvehicleClass in (_faction get "vehiclesDropPod") ) exitWith { 
+   [_markerX] call A3A_fnc_LOG_Crashsite_Satelite;
+};
+
+private _searchHeliClass =  if (_difficult) then {
+    selectRandom ((_faction get "vehiclesHelisLight") +(_faction get "vehiclesHelisLightAttack") + (_faction get "vehiclesHelisAttack"))
+} else {
+    selectRandom ((_faction get "vehiclesHelisLight") + (_faction get "vehiclesHelisLightAttack"))
+};
+private _cargoTruckClass = selectRandom (_faction get "vehiclesTrucks");
+
+private _blackboxClass = "";
+
+///our blackbox
+if ("lowTech" in A3A_factionEquipFlags) then {
+   _blackboxClass = "Land_MetalCase_01_medium_F";
+}else{
+   _blackboxClass = "Land_PortableServer_01_black_F";
+};
+///
+
+private _specOpsArray = if (_difficult) then {selectRandom (_faction get "groupSpecOpsRandom")} else {selectRandom ([_faction, "groupsTierSquads"] call SCRT_fnc_unit_flattenTier)};
+///new
+private _infantrySquadArray = selectRandom ([_faction, "groupsTierMedium"] call SCRT_fnc_unit_flattenTier);
+
+if (isNil "_reconvehicleClass" || 
+    {isNil "_cargoTruckClass"} || 
+    {isNil "_specOpsArray"}) 
+exitWith {
+	["LOG"] remoteExec ["A3A_fnc_missionRequest",2];
+    Error("Problems with faction template, rerequesting new logistics mission.");
+};

```

Comment:
This isn't that readable. It would be better to do it this way since the variable names are quite short and you're not checking much
```suggestion
if (isNil "_reconvehicleClass" || {isNil "_cargoTruckClass"} || {isNil "_specOpsArray"}) exitWith {
	["LOG"] remoteExec ["A3A_fnc_missionRequest",2];
    Error("Problems with faction template, rerequesting new logistics mission.");
};
```